### DISCUSSION TITLE
Enclosure with two fans

### DISCUSSION THREAD

**Discussion by heidepiek** [ale1800/YULC#7] [Ideas]
I know you chose an open enclosure to properly cool the controller. But if we make an enclosure with good airflow and two fans. Would that be enough to keep everything cool? It is still work in progress, but i would like your opinion.

Harrie

![yulc controller proto type](https://github.com/user-attachments/assets/21571720-aba9-442a-9868-0fb6f2eac4ae)
![yulc controller proto type(1)](https://github.com/user-attachments/assets/5b045567-ef7d-4c7a-a763-a23e9dfa26c2)

**Comment by ale1800** [comment]
Hi, the idea is cool! A closed case would be useful in a lot of situations. 
This was my first try:
![PXL_20240623_172348901.jpg](https://github.com/user-attachments/assets/c28f2e40-5413-49c7-b07e-e703ed75539e)

If you want I can share the step file so you can directly modify it.

I think that even one fan could work placed vertically on the side, so moving the heat away from components. 

I made a lot of attempts and I had the best result when the fan was around 0.2/0.3cm above the inductor heatsink.

**Reply by heidepiek** [reply]
Was thinking about that too, fan on the side. Just started drawing. Want to use the controller for my lighting under  my audio cabinet. But prefer the electronics to be somewhat shielded. Hence the idea of a somewhat denser enclosure. Your enclosure looks a bit more compact than mine. I would appreciate getting the step file. Maybe I can get some ideas. Always nice to see how others do it. 

Harrie

**Reply by ale1800** [reply]
I uploaded the step files. Also of yulc itself 👍
[yulc_3D.zip](https://github.com/user-attachments/files/18107837/yulc_3D.zip)

Edit: These step files are of the current "open" case. I don't have the closed one anymore. I think I deleted it

**Reply by heidepiek** [reply]
Ah, no worry`s! ☺️ I continue crafting, when I have made something nice I will show it to you alessandro.

**Comment by robertlipe** [comment]
That looks very cool.  Are they phased so one fan is in and one is out so
you're moving air through the box instead of both fans trying to blow in
(or out) of a sealed box?


Related, I have some changes in the oven for NightDriver that take the
temperature reading inside the ESP32 itself (not the high-power side) and
disable the FET to try to keep everything safe. It's inspired by the mod
that Allesandro uses for WLED.  It probably SHOULD try to send an alert or
something, but I don't have a great way to do that. Maybe just setting the
output to 25% so that it looks distressed instead of broken will help. It's
a little hard to decide how hot is too hot, so I may skip the web setting
and just pick a catastrophic temperature and say that, maybe 40C is just
too hot. Real world numbers welcome.

This is also the kind of change where I've not made a great decision
whether this is best distributed in a YULC-specific build (either "from"
Allesandro or myself) or whether I should try to make it work on all
ESP32's and argue it into the main distribution. I kind of have the same
quandry both in sensing a voltage sag (overloaded output) or a blown fuse
(very overloaded output). There's just not a great way in the project now
to notify a human, but maybe we need one. Does the project need to know how
to send mail or something?

Ideas welcome.

**Reply by heidepiek** [reply]
That was the plan though robert, push pull. But I think I am changing the plan again. And mount the fans on the side. Have read a bit about the temperature sensor of an esp32 S3. It is used to measure the internal temperature of the chip itself. The ambient temperature you need a separate sensor for. So if you want to monitor a fet you will have to use an external sensor for this. I believe the yulc controller also measures the internal temperature right? There is no external sensor mounted anywhere right? i think a safe temperature is 60 degrees. Sure it would be nice if you got a notification on your phone. Personally, I would settle for the controller turning off when the temperature is too high. Which then becomes apparent when the powered LEDs are no longer on. 

☺️

**Comment by robertlipe** [comment]
I haven't seen an external sensor on the schematic
<https://github.com/ale1800/YULC/blob/main/Schematic/yulc_schematic.pdf>.
All we have is what's on the S3 itself which is, admittedly, a bit crude.
So it's not great for controlling your steak's doneness, but it's a
reasonable proxy for "are we on fire?" and that's something we can probably
do at least a little bit about, by shutting down the LED power because if
we're hot, it's going to be from overdriving the power end. 60 degrees? OK.
Sold.

We dont' currently have a way to deliver a notification to your handset. I
don't know of a great way to do so. I sure don't want to get into the
business of building a dedicated app for two OSes.

We can certainly display something in the telnet/serial console and maybe
in the web UI (I don't really do the web UI, but it can't be THAT hard to
just splash some text in red somewhere) I'd thought about cutting the
brightness low as an indicator that something is wrong, but maybe if we
have temps that high just shutting down until we've rebooted or the
temperature is safe (I'm kind of leaning toward staying off just to summon
a human to come check the filters that have a bird nest in them or a small
fire growing in the sealed box) or something is a sane thing to do.


This is one of these things where figuring out what we can and should do is
way higher than doing it. Calling readTemperature once every ten seconds
and comparing it against a constant (maybe something set by the web UI/json
API) and Doing Something is pretty straight forward. It's when "Doing
Something" involves writing mobile apps that it starts to turn into real
work.

I'm kind of a fan of NOT bursting into flames if we can avoid it. It's bad
press. I don't think we have to solve this perfectly to solve it "well
enough".

"Not all things worth doing are worth doing well."

If we do nothing but sense that it's crazy hot, display a message, and turn
off the LEDs, that's a gold star, IMO. I'll keep poking at it.

RJL

**Comment by ale1800** [comment]
Yes, there are no external temperature sensors. Only the S3 internal one. But it can follow well enough the inductor/board temperature on the other side. It's obviously not perfect but it's a good guess. What I do with the internal temperature usemod in wled is just setting strip brightness to 0, turning it off. It's handy because you don't have to care if you have an analog strip and an addressable one connected together, or two analog ones. The software handles the way of how to turn off that type of strip. 

In my opinion ND should do the same thing. Considering that for the addressable one you should turn off through "software" before turning off the mosfet, so it's slightly different from the common "relay" solution that cut the main.
Or even better, checking the temperature every **random** seconds and decreasing the brightness until the board is not "cold" enough. During my tests I could see that when you decrease the brightness the temperature goes down almost instantly.

**Reply by heidepiek** [reply]
Trying my best, this is a prototype to see if all would fit. I used petg,  the roll filament was already sitting a couple off weeks on the ender. So the print quality is not at best (hydroscopic)😰. I think my ender can do better!? But here is an example😜 

![yulc enclosure proto](https://github.com/user-attachments/assets/671f68ea-177c-4a61-9afd-5d1b191bd403)

There is a cutout for a inmp441 mems microphone so you can use the audioreactive feature of wled😏 

Greetz from holland!

![yulc controller 1](https://github.com/user-attachments/assets/5b6eecc0-1fa8-4a70-87ed-86a541118ad0)
![yulc controller 2](https://github.com/user-attachments/assets/1d2c00fd-0cc0-43d3-b6a3-e96cb6660341)
![yulc controller 3](https://github.com/user-attachments/assets/5766908b-a60f-4dcf-918d-2d6dfd3b9e90)

**Reply by ale1800** [reply]
Cool!  Were you able to re-use the M2 screws to lock in place the board?

**Reply by heidepiek** [reply]
What do you think alessandro? Is this enclosure worthy of the yulc controller? Would you like the stl files to see for yourself if everything is good? Are still a few things I want to change myself to make the inserts fit better for example. Sometimes you have to enlarge some holes with a drill bit. And I forgot the boot button in the housing I am showing now. But I have already changed that in the stl file. 

https://drive.google.com/file/d/1Yt43t7V7vUki6mSefxsGJrpgrXxnkq8-/view?usp=sharing

Regards Harrie 🙋

**Reply by heidepiek** [reply]
I used small self-tapping screws to secure the motherboard. But you can burn in inserts if necessary. Did use inserts for the small feet for under the case. And also used inserts for the two screw holes for the lid. I have way too much stuff lying around alessandro!🙈  Shall I just put the stl files here!  If people want to change something they can modify it in tinkercad. That's where I fabricated all this too! 

![yulc enclosure proto](https://github.com/user-attachments/assets/ed79f926-52bf-4693-84de-cc5f8c097a36)
![yulc enclosure proto4](https://github.com/user-attachments/assets/39da0fc9-034d-4cc9-9e4b-5c555083d85b)
![yulc enclosure proto3](https://github.com/user-attachments/assets/16b68932-6c99-448d-a392-34fc9aac0c3b)
![yulc enclosure proto2](https://github.com/user-attachments/assets/14dde499-a17f-48d3-85fe-954944423a31)
![yulc enclosure proto1](https://github.com/user-attachments/assets/4f51e943-4bce-4e02-a817-7bbe50fb8c77)



Printed everything in petg due to heat from the yulc controller. But turned out well in my opinion! Try it alessandro.


[yulc controller stl files.zip](https://github.com/user-attachments/files/18140269/yulc.controller.stl.files.zip)

😉

**Reply by ale1800** [reply]
It looks great in the video! I only have some doubts. You'll need a really long screwdriver to use the screw terminals, is it correct? And how would you connect the two fans to yulc? 
I think you would need some soldering re-work.

**Reply by heidepiek** [reply]
I like to tinker, hope that i can take some doubts away. The two cables, yes you have to solder two cables. 
Will show you for reference that you do not need a very long screwdriver for the terminals. I Use an ordinary precision screwdriver. It really is a very nice compact housing. I think you should print it first alessandro. You can also just print it from pla. Doesn't cost that much filament. You can see it better in real life. As I said, someone who can tinker a little doesn't turn his hand around for those two cables. It won't be about that. 



![yulc housing3](https://github.com/user-attachments/assets/c13cd334-0bc6-4a3f-b51d-6f3f89ebc407)
![yulc housing](https://github.com/user-attachments/assets/d73c3ef3-5d5e-4f53-9e87-15afa9ce3d27)
![yulc housing2](https://github.com/user-attachments/assets/dba2eee3-b785-4954-b93a-d21da3fa7ae6)

Harrie😉

**Reply by heidepiek** [reply]
Still having doubts alessandro? Really, the only thing is that you have to put the two cables together😜  . You can just use a small screwdriver to tighten the terminal screws. Someone who really wants your controller in a sealed enclosure will get that done💪 .  But I wasted another evening making a proto type😏 (sarcasm).  How does this look to you?

![yulc prototype 2](https://github.com/user-attachments/assets/ebdf1838-60e4-45d2-85e2-92430e7bf282)
![yulc prototype 2(1)](https://github.com/user-attachments/assets/d3e52ffa-622d-4b22-8f52-520ebfc00026)
![yulc prototype 2(2)](https://github.com/user-attachments/assets/8dfb91e6-241f-4be4-af7d-611ce86ffe6f)

☺️

**Reply by ale1800** [reply]
This one seems better! The fan on the side it's the right choice and the top part looks great!

Can you share both the stl and the step files? 

I would like to print and try to modify some small things 👍

**Reply by heidepiek** [reply]
Unfortunately, I can’t provide STEP files, as I created everything in Tinkercad, which doesn’t support that format. You could try importing an STL file into a CAD program and converting it to STEP, though I don’t have experience with that myself.

I don’t have a background in graphic design or electronics, but our Maker has blessed me with a pair of good hands—what my eyes see, my hands can build (most of the time).

As I write this, I’m still working on the second prototype. The only reason I’m pushing through is because you prefer this housing design. Otherwise, I might not have finished it, since I’m already satisfied with the first design. Everything fits well, functions properly, and came out nicely from the 3D printer. I printed it in PETG, which can be tricky to work with at times, but I chose it because the Yulc controller can get warm, and PETG handles heat better than PLA.

It’s interesting how opinions can differ. The design with the two fans worked well for me. What made you hesitate about that design? The longer screwdriver argument doesn’t really apply—you can use a regular precision screwdriver. Connecting the two fans isn’t rocket science either.

Anyway, I’ll finish the housing so I can share the STL files with you. It’s taken quite a few hours, but hey—that’s the hobby! If you like it, print it; if not, feel free to modify it!

You did an excellent job with the controller, Alessandro! I hope you’ll find this housing useful.

Best regards,
Harrie

**Reply by heidepiek** [reply]
Spent another two hours tinkering, and this is the final result. I hope you can make good use of it, Alessandro.

This time, I didn’t print the housing, so I hope everything fits well. If not, you might need to make some adjustments. Maybe I’ll print this version myself at some point, but not this week. There are plenty of other things piling up, so I need to set some priorities (my wife also wants me to do more than just hang out in the man cave!)🙈 .



✌️

**Reply by ale1800** [reply]
Thanks! I'm sure the one with two fans works great. But I'm giving only one fan, so you should buy another one and then soldering both together. It's not so difficult but it's also not obvious that everyone want/can do that. I would prefer something simpler and straight forward like the one with the fan on the side. 
And I think that it could move more air compared to the fan on the top. As I said I tried with the fan on the top and I didn't have great results pushing the output current to the limit (but it was only one fan).

So I'll definitely try the side-version, I'll let you know!

**Comment by robertlipe** [comment]
I think I know both of you somewhat and may understand the differing
perspective. I've enjoyed reading the volley here!

A *project* and a *product* sound similar, but they're different.

Harrie is building a project *for Harrie*. Quantity 1. Maybe 2. It can use
whatever parts, filaments, connectors, and tools Harrie has. It will expand
to fill whatever amount of tinker time is allotted to it in the man cave of
much music.

Allesandro is trying to offer something to (hopefully) thousands of buyers
that may have minimal tools and skills. They may not own a soldering iron.
These people may send a files to a friend, a library, or a service that
does these things; exchange some beer, beaver pelts, coins, or other
trades; and hope to assemble it into something useful using the unmodified
fan included with the YULC. Adding the cost of an extra fan to everything
shipped adds up.


I also think I remember that Alessandro has one of those fancy FLIR cameras
that can measure exactly how effective cooling is in real time. I know he
has resistive dummy loads that can drive that regulation circuit to the
limit and can check individual component temps against data sheets and
lifecycle deregulation charts and all that. I'm pretty sure I recall from
my days at a hardware company that a crosswind "breeze" to carry air over
components was strongly preferred to blowing in and out from the top of a
PCB.

This reminds me. I think I lost the beat with you, Allesandro. I have
had some personal distractions lately. I need to find those boards and get
back to that. Sorry!

**Reply by heidepiek** [reply]
Wise words robert😉, in my subconscious I knew this too. But I was so focused on making a nice enclosure. Printed the enclosure (couldn't resist) to see if everything fit. Hope I was earlier than alessandro. The lid is a bit on the tight side! Think it would be better fixed with 4 bolts! It does fit, but not to my liking. Tip for alessandro, you had bolts included with the controller. Printed your housing too, and when I wanted to use your bolts I couldn't tighten them. The holes were a little too big. Now there are 100 and 1 printers made. And not all printers print the same. Also mainly has to do with adjustment. I would personally supply a few self-tapping screws. If the hole is too small you can always drill it.  Think you are also skilled enough to adjust the housing, but I'm also  going to do that.

😌 
![proto2](https://github.com/user-attachments/assets/e51c1bef-0bab-4e98-863d-11879e12a14c)

Update: 

I have adjusted the lid and the housing to have four holes. I’ve also printed everything, and now the lid fits nicely on the housing. So, I can, in good conscience, share the STL files here.

[housing 4 screws.zip](https://github.com/user-attachments/files/18185584/housing.4.screws.zip)

![4 screws lid](https://github.com/user-attachments/assets/5658e159-50d3-4368-8959-accd3440453a)

**Comment by heidepiek** [comment]
Am curious alessandro, have you printed the enclosure yet? And testing how it goes with heat buildup and dissipation? 

Harrie :)

**Reply by ale1800** [reply]
Hi Harrie, sorry for the late response. I've not printed it yet, I've been busy with other things after the holidays. I'll soon catch up with this 👍
